tests/annotator/annotator.cpp

```diff
@@ -1970,78 +2022,64 @@ TEST(Annotator, autoIdOnOutOfDateBuild)
     auto model = parser->parseModel(modelStringUniqueIds);
     auto annotator = libcellml::Annotator::create();
 
-    annotator->buildModelIndex(model);
-    EXPECT_TRUE(annotator->isBuilt());
+    annotator->setModel(model);
 
     model->component(0)->setId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da55", annotator->assignId(libcellml::Annotator::Type::COMPONENT, model->component(0)));
+    EXPECT_EQ("b4da55", annotator->assignId(model->component(0), libcellml::CellMLElement::COMPONENT));
     EXPECT_EQ("b4da55", model->component(0)->id());
 
     model->component("component2")->setEncapsulationId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da56", annotator->assignId(libcellml::Annotator::Type::COMPONENT_REF, model->component("component2")));
+    EXPECT_EQ("b4da56", annotator->assignId(model->component("component2"), libcellml::CellMLElement::COMPONENT_REF));
     EXPECT_EQ("b4da56", model->component("component2")->encapsulationId());
 
     libcellml::VariablePair connection = std::make_pair(
         model->component("component2")->variable("variable1"),
         model->component("component2")->variable("variable1")->equivalentVariable(0));
     libcellml::Variable::setEquivalenceConnectionId(connection.first, connection.second, "changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da57", annotator->assignId(libcellml::Annotator::Type::CONNECTION, connection));
+    EXPECT_EQ("b4da57", annotator->assignId(connection));
```

Comment:
I think there shouldn't be a default for variable pair items (or if there is, it should be the map_variables one).  But better without any.